cmd/kubectl-direct_csi/drives_format.go

```diff
@@ -83,100 +81,68 @@ $ kubectl direct-csi drives format <drive_id_1> <drive_id_2>
-		}
-	} else {
-		driveList, err := directClient.DirectCSIDrives().List(ctx, metav1.ListOptions{})
-		if err != nil {
-			return err
-		}
-
-		if len(driveList.Items) == 0 {
-			klog.Errorf("No resource of %s found\n", bold("DirectCSIDrive"))
-			return fmt.Errorf("No resources found")
-		}
-
-		accessTierSet, aErr := getAccessTierSet(accessTiers)
-		if aErr != nil {
-			return aErr
+	wg := sync.WaitGroup{}
+	accessTierSet, aErr := getAccessTierSet(accessTiers)
+	if aErr != nil {
+		return aErr
+	}
+	for d := range driveCh {
+		if !d.MatchGlob(nodes, drives, status) {
+			continue
 		}
 
-		for _, d := range driveList.Items {
-			if d.MatchGlob(nodes, drives, status) {
-				if d.MatchAccessTier(accessTierSet) {
-					filterDrives = append(filterDrives, d)
-				}
-			}
+		if !d.MatchAccessTier(accessTierSet) {
+			continue
 		}
-	}
-
-	driveName := func(val string) string {
-		dr := strings.ReplaceAll(val, sys.DirectCSIDevRoot+"/", "")
-		dr = strings.ReplaceAll(dr, sys.HostDevRoot+"/", "")
-		return strings.ReplaceAll(dr, "-part-", "")
-	}
 
-	for _, d := range filterDrives {
 		if d.Status.DriveStatus == directcsi.DriveStatusUnavailable {
 			continue
 		}
 
+		path := canonicalNameFromPath(d.Status.Path)
+		nodeName := d.Status.NodeName
+		driveAddr := fmt.Sprintf("%s:/dev/%s", nodeName, path)
+
 		if d.Status.DriveStatus == directcsi.DriveStatusInUse {
-			driveAddr := fmt.Sprintf("%s:/dev/%s", d.Status.NodeName, driveName(d.Status.Path))
-			klog.Errorf("%s in use. Cannot be formatted", utils.Bold(driveAddr))
+			klog.Errorf("%s is in use. Cannot be formatted",
+				utils.Bold(driveAddr))
 			continue
 		}
 
-		if d.Status.DriveStatus == directcsi.DriveStatusReady {
-			driveAddr := fmt.Sprintf("%s:/dev/%s", d.Status.NodeName, driveName(d.Status.Path))
-			klog.Errorf("%s already owned and managed. Use %s to overwrite", utils.Bold(driveAddr), utils.Bold("--force"))
+		if d.Status.DriveStatus == directcsi.DriveStatusReady && !force {
+			klog.Errorf("%s is already owned and managed. Use %s to reformat",
+				utils.Bold(driveAddr), utils.Bold("--force"))
 			continue
 		}
 		if d.Status.Filesystem != "" && !force {
-			driveAddr := fmt.Sprintf("%s:/dev/%s", d.Status.NodeName, driveName(d.Status.Path))
-			klog.Errorf("%s already has a fs. Use %s to overwrite", utils.Bold(driveAddr), utils.Bold("--force"))
+			klog.Errorf("%s already has a fs. Use %s to overwrite",
+				utils.Bold(driveAddr), utils.Bold("--force"))
 			continue
 		}
 
 		if d.Status.DriveStatus == directcsi.DriveStatusReleased {
-			if !unrelease {
-				driveAddr := fmt.Sprintf("%s:/dev/%s", d.Status.NodeName, driveName(d.Status.Path))
-				klog.Errorf("%s is in 'released' state. Use %s to overwrite and make it 'ready'", utils.Bold(driveAddr), utils.Bold("--unrelease"))
-				continue
-			}
-			d.Status.DriveStatus = directcsi.DriveStatusAvailable
+			klog.Errorf("%s is in 'released' state. Use 'kubectl direct-csi unrelease --drive %s --nodes %s' before formatting",
```

Comment:
There is no `unrelease` command. We have to bring back the `--unrelease` flag to make these drives ready again.